bazel/curl.BUILD

```diff
@@ -55,5 +57,9 @@ cc_library(
             "-lpthread",
         ],
     }),
+    deps = [
+        "@boringssl//:crypto",
+        "@boringssl//:ssl",
+    ],
```

Comment:
The CMake build has a dependency on OpenSSL. Can we have the same dependency to maintain consistency?

Reply:
Agree, but gRPC also use boringssl, maybe we can use alias to switch the ssl library. (https://bazel.build/docs/configurable-attributes?#faq-select-bind)